pkgs/tools/security/gnupg/24.nix

```diff
@@ -1,13 +1,17 @@
-{ fetchurl, fetchpatch, lib, stdenv, pkg-config, libgcrypt, libassuan, libksba
-, libgpg-error, libiconv, npth, gettext, texinfo, buildPackages
-, nixosTests
-, guiSupport ? stdenv.isDarwin, enableMinimal ? false
-, adns, bzip2, gnutls, libusb1, openldap
-, pinentry, readline, sqlite, zlib
+{ lib, stdenv, fetchurl, buildPackages
+, pkg-config, texinfo
+, gettext, libassuan, libgcrypt, libgpg-error, libiconv, libksba, npth
+, adns, bzip2, gnutls, libusb1, openldap, readline, sqlite, zlib
+, enableMinimal ? false
 , withPcsc ? !enableMinimal, pcsclite
+, guiSupport ? stdenv.isDarwin, pinentry
 , withTpm2Tss ? !stdenv.isDarwin && !enableMinimal, tpm2-tss
 }:
 
+let
+  pinentryBinaryPath = pinentry.binaryPath or "bin/pinentry";
+in
+
```

Comment:
```suggestion
```

Reply:
I think this is a remnant of the proposal to have a `common.nix` used by `22.nix` and `24.nix`. Since the preference is avoid `common.nix` it makes sense to inline this as you are suggesting below 👍

Reply:
I'm too in favor of `common.nix` along with `default.nix` and `lts.nix` or alike BTW. I don't expect upstream to change their build system that much that and I expect the deviations between the expressions to be small.

Reply:
Thanks for chiming in on this, @doronbehar. I'd be happy to re-add the `common.nix` if the maintainers or a majority of nixpkgs developer / contributors speaks in favor of it. If I remember correctly there are less than a handful for and against it :)